protected/models/User.php

```diff
@@ -482,11 +482,14 @@ public function beforeValidate()
 
         if (!$this->global_firm_rights){
             $request = Yii::app()->getRequest();
+            $firms = null;
             if ($request->getIsPostRequest()){
                 $user = $request->getPost('User');
                 Yii::log(var_export($user, true));
             }
-            $firms = $user['firms'];
+            if(isset($user['firms'])){
+                $firms = $user['firms'];
+            }
```

Comment:
I'm not sure how safe this block is. I think the following block would work better:

```$firms = isset($user['firms']) ? $user['firms'] : null;```